core/shared/mem-alloc/ems/ems_gc.c

```diff
@@ -378,8 +379,8 @@ reclaim_instance_heap(gc_heap_t *heap)
                     offset = ref_start_offset + j * sizeof(void *);
                     bh_assert(offset + sizeof(void *) < size);
                     ref = *(gc_object_t *)(((gc_uint8 *)obj) + offset);
-                    if (ref == NULL_REF)
-                        continue; /* NULL REF */
+                    if (ref == NULL_REF || ((uintptr_t)ref & 1))
```

Comment:
I think i31ref should not be marked as reference? So it's not possible for i31ref to come here?

Reply:
The i31 ref in GC AOT compilation is pushed to stack as a GC REF and will be committed to stack frame. And the operands in params/locals/stacks whose type are anyref or eqref can also be i31refs, they may be committed to stack frame also, in the compilation time, the AOT compiler is not easy to know that they are i31ref and doesn't commit ref flag to stack frame.

Reply:
Got it, thanks!